controllers/versioneddataset_controller.go

```diff
@@ -44,7 +44,7 @@ type VersionedDatasetReconciler struct {
 	cache sync.Map
 }
 
-//+kubebuilder:rbac:groups=arcadia.kubeagi.k8s.com.cn,resources=versioneddatasets,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=arcadia.kubeagi.k8s.com.cn,resources=versioneddatasets,verbs=get;list;watch;create;update;patch;delete;deletecollection
 //+kubebuilder:rbac:groups=arcadia.kubeagi.k8s.com.cn,resources=versioneddatasets/status,verbs=get;update;patch
```

Comment:
Since we use this `deletecollection` at https://github.com/kubeagi/arcadia/blob/main/controllers/dataset_controller.go#L64 

Would it be better to place this in dataset controller?

Reply:
- //+kubebuilder:rbac:groups=arcadia.kubeagi.k8s.com.cn,resources=datasets,verbs=get;list;watch;create;update;patch;delete;deletecollection

- //+kubebuilder:rbac:groups=arcadia.kubeagi.k8s.com.cn,resources=versioneddatasets,verbs=deletecollection

The first one means adding the delete collection permissions to  dataset.
The second means adding delete collection permissions to versioneddataset.

So it's wrong to write directly to the dataset, if we use the second way of writing, then I think it's fine to add directly to the controller of the `versioneddataset`, with permissions to the same resource all written together.

Reply:
Well... of source we need this deletecollection permission to `//+kubebuilder:rbac:groups=arcadia.kubeagi.k8s.com.cn,resources=versioneddatasets,verbs=deletecollection`

It's fine you add `deletecollection` in this file since the permission will be granted to manager-role too. 

But from my understanding, this `//+kubebuilder:rbac:groups=arcadia.kubeagi.k8s.com.cn,resources=versioneddatasets,verbs=deletecollection` is the way we request permission for current controller. So 

request it where you where you think should use this permission

Reply:
Either way is good to me. Please resolve the conlicts